app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt

```diff
@@ -0,0 +1,413 @@
+                DIFF_TYPE_LINE_WITH_SAME_CONTENT -> {
+                }
+                DIFF_TYPE_LINE_ADDED -> {
+                    updateDiffTextDecor(spannableString, true, prefixLength, prefixLength + diff.text.length)
+                }
+                DIFF_TYPE_LINE_REMOVED -> {
+                    updateDiffTextDecor(spannableString, false, prefixLength, prefixLength + diff.text.length)
+                }
+                DIFF_TYPE_LINE_WITH_DIFF -> {
+                    for (highlightRange in diff.highlightRanges) {
+                        val highlightRangeStart = if (languageCode == "en") highlightRange.start
+                        else getByteInCharacters(diff.text, highlightRange.start, 0)
+                        val highlightRangeLength = if (languageCode == "en") highlightRange.length
+                        else getByteInCharacters(diff.text, highlightRange.length, highlightRangeStart)
+
+                        if (highlightRange.type == HIGHLIGHT_TYPE_ADD) {
+                            updateDiffTextDecor(spannableString, true, prefixLength + highlightRangeStart, prefixLength + highlightRangeStart + highlightRangeLength)
+                        } else {
+                            updateDiffTextDecor(spannableString, false, prefixLength + highlightRangeStart, prefixLength + highlightRangeStart + highlightRangeLength)
+                        }
+                    }
+                }
+                DIFF_TYPE_PARAGRAPH_MOVED_FROM -> {
+                    updateDiffTextDecor(spannableString, false, prefixLength, prefixLength + diff.text.length)
+                }
+                DIFF_TYPE_PARAGRAPH_MOVED_TO -> {
+                    updateDiffTextDecor(spannableString, true, prefixLength, prefixLength + diff.text.length)
+                }
+            }
+            diffText.text = spannableString.append("\n")
+        }
+    }
+
+    private fun updateDiffTextDecor(spannableText: SpannableStringBuilder, isAddition: Boolean, start: Int, end: Int) {
+        val boldStyle = StyleSpan(Typeface.BOLD)
+        val foregroundAddedColor = ForegroundColorSpan(ResourceUtil.getThemedColor(requireContext(), R.attr.color_group_64))
+        val foregroundRemovedColor = ForegroundColorSpan(ResourceUtil.getThemedColor(requireContext(), R.attr.color_group_65))
+        spannableText.setSpan(BackgroundColorSpan(ResourceUtil.getThemedColor(requireContext(),
+                if (isAddition) R.attr.edit_green_highlight else R.attr.edit_red_highlight)), start, end, 0)
+        spannableText.setSpan(boldStyle, start, end, 0)
+        spannableText.setSpan(if (isAddition) foregroundAddedColor else foregroundRemovedColor, start, end, 0)
+    }
+
+    private fun getByteInCharacters(diffText: String?, byteLength: Int, start: Int): Int {
+        var charCount = 0
+        var bytes = byteLength
+
+        for (pos in start until diffText!!.length - 1) {
+            val idBytes = diffText[pos].toString().toByteArray(CHARSET).size
```

Comment:
This is using something called `CHARSET`, which comes from `com.bumptech.glide.load.Key`, which seems like a weird dependency.  Can we just use `StandardCharsets.UTF_8`?